api/app/Http/Controllers/Api/Auth/EmailVerificationController.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Controllers\Api\Auth;
+
+use App\Actions\Auth\EmailVerificationAction;
+use App\Actions\Auth\EmailVerificationRequest;
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+
+class EmailVerificationController extends Controller
+{
+    public function verify(Request $request, EmailVerificationAction $action)
+    {
+        $emailVerificationRequest = new EmailVerificationRequest(
+            $request->id,
+            $request->hash
+        );
+
+        return $action->execute($emailVerificationRequest);
```

Comment:
You should not directly return the response from the action to the client. In this case, the response is not needed so you can return empty one.